applications/fishing-map/src/features/analysis/Analysis.tsx

```diff
@@ -0,0 +1,243 @@
+  const analysisQuery = useSelector(selectAnalysisQuery)
+  const [sourceLoaded, setSourceLoaded] = useState(false)
+  const timeoutRef = useRef<NodeJS.Timeout>()
+  const { dispatchQueryParams } = useLocationConnect()
+  const { updateFeatureState, cleanFeatureState } = useFeatureState(mapInstance)
+  const staticTime = useSelector(selectStaticTime)
+  const dataviews = useSelector(selectTemporalgridDataviews) || []
+  const analysisGeometry = useSelector(selectAnalysisGeometry)
+  const reportStatus = useSelector(selectReportStatus)
+  const userData = useSelector(selectUserData)
+  const hasAnalysisLayers = useSelector(selectHasAnalysisLayersVisible)
+  const { areaId, sourceId } = analysisQuery
+
+  useLayoutEffect(() => {
+    // We have to go to zoom 0 to load every geometry to be able to fitBounds later
+    const viewport = { latitude: 0, longitude: 0, zoom: 0 }
+    setMapCoordinates(viewport)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
+
+  useEffect(() => {
+    const sourceCallback = () => {
+      if (mapInstance?.getSource(sourceId) && mapInstance?.isSourceLoaded(sourceId)) {
+        setSourceLoaded(true)
+        mapInstance.off('idle', sourceCallback)
+      }
+    }
+
+    if (mapInstance && sourceId && areaId) {
+      setSourceLoaded(false)
+      mapInstance.on('idle', sourceCallback)
+    }
+
+    return () => {
+      if (mapInstance) {
+        mapInstance.off('idle', sourceCallback)
+      }
+    }
+  }, [mapInstance, sourceId, areaId])
+
+  useEffect(() => {
+    if (sourceLoaded && mapInstance) {
+      const contextAreaFeatures = mapInstance?.querySourceFeatures(sourceId, {
+        sourceLayer: 'main', // Our layers always uses 'main' as the source layer
+        filter: ['==', 'gfw_id', parseInt(areaId)],
+      })
+      if (contextAreaFeatures.length > 0) {
+        cleanFeatureState('highlight')
+        const contextAreaFeatureMerged = contextAreaFeatures.reduce(
+          (acc, { geometry, properties }) => {
+            const featureGeometry: Feature<Polygon> = {
+              type: 'Feature',
+              geometry: geometry as Polygon,
+              properties,
+            }
+            if (!acc?.type) return featureGeometry
+            return union(acc, featureGeometry, { properties }) as Feature<Polygon>
+          },
+          {} as Feature<Polygon>
+        )
+        const { name, value } = contextAreaFeatureMerged.properties || {}
+        dispatch(
+          setAnalysisGeometry({
+            geometry: contextAreaFeatureMerged,
+            name: name || value,
+          })
+        )
+        const bounds = bbox(contextAreaFeatureMerged) as [number, number, number, number]
+        fitMapBounds(bounds, 10)
+        const featureState = { source: sourceId, sourceLayer: 'main', id: areaId }
+        updateFeatureState([featureState], 'highlight')
+      } else {
+        console.warn('No feature for analysis found')
```

Comment:
How should the user perceive this scenario? is it possible? what should we display to them?

Reply:
It shouldn't ever happen... in the current state because you can't remove any layer from the analysis but thanks for the catch! I've handled the error in case this happens using this:
<img width="384" alt="image" src="https://user-images.githubusercontent.com/10500650/110118127-2bf81600-7dba-11eb-8508-be34018b3690.png">